exp/zapslog/slog.go

```diff
@@ -157,33 +213,52 @@ func (h *Handler) Handle(ctx context.Context, record slog.Record) error {
 		}
 	}
 
-	fields := make([]zapcore.Field, 0, record.NumAttrs())
+	fields := _fieldSlices.Get()
+	defer fields.Return()
+
 	record.Attrs(func(attr slog.Attr) {
-		fields = append(fields, convertAttrToField(attr))
+		converted := convertAttrToFields(attr)
+		fields.Extend(converted)
+		converted.Return()
 	})
-	ce.Write(fields...)
+	ce.Write(*fields...)
+
 	return nil
 }
 
 // WithAttrs returns a new Handler whose attributes consist of
 // both the receiver's attributes and the arguments.
 func (h *Handler) WithAttrs(attrs []slog.Attr) slog.Handler {
-	fields := make([]zapcore.Field, len(attrs))
-	for i, attr := range attrs {
-		fields[i] = convertAttrToField(attr)
+	fields := _fieldSlices.Get()
+	defer fields.Return()
+
+	for _, attr := range attrs {
+		converted := convertAttrToFields(attr)
+		fields.Extend(converted)
+		converted.Return()
 	}
-	return h.withFields(fields...)
+
+	return h.withFields(fields)
 }
 
 // WithGroup returns a new Handler with the given group appended to
 // the receiver's existing groups.
 func (h *Handler) WithGroup(group string) slog.Handler {
-	return h.withFields(zap.Namespace(group))
+	// Stack-allocate here: the number of options is known at compile time.
+	tmp := [...]zapcore.Field{
+		zap.Namespace(group),
+	}
+
+	cloned := *h
+	cloned.core = h.core.With(tmp[:])
```

Comment:
I'm 75% certain that the manual allocation won't have any effect.
It will behave the same as `h.core.With(zap.Namespace(group))`:
if the slice escapes, it'll be heap allocated anyway, and if it doesn't escape, it was going to be stack allocated anyway.